python/metatensor-torch/metatensor/torch/atomistic/openmm_interface.py

```diff
@@ -0,0 +1,235 @@
+
+    if not HAS_OPENMM:
+        raise ImportError(
+            "Could not import openmm and/or nnpops. If you want to use metatensor with "
+            "openmm, please install openmm-torch and nnpops with conda."
+        )
+
+    model = load_atomistic_model(path, extensions_directory=extensions_directory)
+
+    # Get the atomic numbers of the ML region.
+    all_atoms = list(topology.atoms())
+    atomic_numbers = [atom.element.atomic_number for atom in all_atoms]
+
+    if atoms is None:
+        selected_atoms = None
+    else:
+        selected_atoms = Labels(
+            names=["system", "atom"],
+            values=torch.tensor(
+                [[0, selected_atom] for selected_atom in atoms],
+                dtype=torch.int32,
+            ),
+        )
+
+    class MetatensorForce(torch.nn.Module):
+
+        def __init__(
+            self,
+            model: torch.jit._script.RecursiveScriptModule,
+            atomic_numbers: List[int],
+            selected_atoms: Optional[Labels],
+            check_consistency: bool,
+        ) -> None:
+            super(MetatensorForce, self).__init__()
+
+            self.model = model
+            self.register_buffer(
+                "atomic_numbers", torch.tensor(atomic_numbers, dtype=torch.int32)
+            )
+            self.evaluation_options = ModelEvaluationOptions(
+                length_unit="nm",
+                outputs={
+                    "energy": ModelOutput(
+                        quantity="energy",
+                        unit="kJ/mol",
+                        per_atom=False,
+                    ),
+                },
+                selected_atoms=selected_atoms,
+            )
+
+            requested_nls = self.model.requested_neighbor_lists()
+            if len(requested_nls) > 1:
+                raise ValueError(
+                    "The model requested more than one neighbor list. "
+                    "Currently, only models with a single neighbor list are supported "
+                    "by the OpenMM interface."
+                )
+            elif len(requested_nls) == 1:
+                self.requested_neighbor_list = requested_nls[0]
+            else:
+                # no neighbor list requested
+                self.requested_neighbor_list = None
+
+            self.check_consistency = check_consistency
+
+        def _attach_neighbors(self, system: System) -> System:
+
+            if self.requested_neighbor_list is None:
+                return system
+
+            cell: Optional[torch.Tensor] = None
+            if not torch.all(system.cell == 0.0):
+                cell = system.cell
+
+            # Get the neighbor pairs, shifts and edge indices.
+            neighbors, interatomic_vectors, _, _ = getNeighborPairs(
+                system.positions,
+                self.requested_neighbor_list.engine_cutoff("nm"),
+                -1,
+                cell,
+            )
+            mask = neighbors[0] >= 0
```

Comment:
will there negative values in `neighbors[0]` if `n_pairs=-1` above?

Reply:
Yes, I think that NL returns a huge Tensor with all pairs, where only a certain number are not -1, i.e., masked

Reply:
I understand this behavior if we are setting `max_n_pairs=300`: if there are fewer pairs, the other ones are filled with -1. You are saying this is also the case when giving `max_n_pairs=-1`?

Reply:
Yes, I tried a few tests (with `max_n_pairs=-1`) and the mask is still necessary. I think they're always returning a number of pairs that is a multiple of 4. E.g. if you have 27 pairs, 28 will be returned and one of them is masked with center=neighbor=-1